zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -675,7 +675,7 @@ object ZClient {
             case Some(Scheme.HTTPS) => Scheme.WSS
             case Some(Scheme.WS)    => Scheme.WS
             case Some(Scheme.WSS)   => Scheme.WSS
-            case None               => Scheme.WS
+            case _                  => Scheme.WS
```

Comment:
I am not sure, that this is correct. Having no `Schema`, it is reasonable to assume `WS`, but having an explicit non websocket schema, is an error I'd say. So I would keep `None => Schema.WS` and fail with an exception else.

Reply:
So -+ like
```suggestion
            case None               => ZIO.succeed(Scheme.WS)
            case _                  => ZIO.fail
```

Reply:
You can not use zio here. The method expects a pure value to be returned. I am not sure, where this method is used internally. Maybe it is okay to throw an exception. Maybe we need to change the return type to option.

Reply:
Yes, I think returning ZIOs is correct here.

Reply:
In your opinion what Throwable should I return in `ZIO.fail`?